### PR TITLE
Add benchmark 

### PR DESCRIPTION
**Pull request by natir**
I create this pull request to track the changes needed to set up a niffler benchmark and discuss these changes. #25 

Main todolist:
- [X] bench format detection
- [X] bench reading of small stream in memory
- [x] bench writting of small stream in memory
- [x] bench reads of a file on disk
- [x] bench writting of a file on disk

Side effect todolist
- [x] write documentation of read_compression function

### CODE REVIEW

**Comment by natir** [comment]
I had to make the format detection function `read_compression` public, I'll have to write its documentation and maybe change its name to `compression_detection` @luizirber do you have an opinion about that name?

**Comment by natir** [comment]
To change where the file is write on disk set environement variable TMPDIR

File: benches/benchmarks.rs [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+Originally from https://github.com/natir/yacrd/blob/3fc6ef8b5b51256f0c4bc45b8056167acf34fa58/src/file.rs
+*/
```
Comment (luizirber) [inline benches/benchmarks.rs]: I think we can remove this copyright notice, since it's in the LICENSE already?

File: src/compression.rs [hunk 2]
```diff
@@ -75,7 +75,7 @@ fn get_first_five<'a>(
     }
 }
 
-pub(crate) fn read_compression<'a>(
+pub fn read_compression<'a>(
```
Comment (luizirber) [inline src/compression.rs]: a cheeky suggestion is calling this `sniff` instead. You called it `detect_format` in the benchmarks, which I think it is also better than `read_compression`.

**Review by luizirber — CHANGES_REQUESTED** [review]
2 small comments, but otherwise looks amazing!

File: src/compression.rs (hunk 2 above)
Comment (luizirber) [reply src/compression.rs]: But also... this ends up being really similar to `get_reader` (but with return values order switched), do you have other use cases to make it public?

File: benches/benchmarks.rs (hunk 1 above)
Comment (natir) [reply benches/benchmarks.rs]: Ok, maybe I can remove all header after this branch is merge.

File: src/compression.rs (hunk 2 above)
Comment (natir) [reply src/compression.rs]: I like the name `sniff`.

In fact `sniff` and `get_reader`are different:
- `sniff` return compression fomat and the raw stream
- `get_reader` return the uncompress stream and the compression format.

So isn't the same function but ok I can change the order of return in `sniff` to have a cleaner interface.

**Comment by natir** [comment]
@luizirber If it's ok for you I think we can merge.

In think, the add of `sniff` in public API didn't break compatibility with the previous version so if we want create a new release the version number is 2.1.0. I think we don't need to create a new release for this little change.

**Comment by natir** [comment]
If you want correct typo I write in the documentation please do it :)

**Review by luizirber — APPROVED** [review]

**Comment by luizirber** [comment]
@natir do the changes in https://github.com/luizirber/niffler/pull/25/commits/025c7c7411da17457b2504ba4878306f9920e998 look good? If yes, feel free to merge =]

**Pull request merged**